pandas/core/indexes/base.py

```diff
@@ -2378,12 +2378,49 @@ def identical(self, other):
 
     def asof(self, label):
         """
-        For a sorted index, return the most recent label up to and including
-        the passed label. Return NaN if not found.
+        Return the latest index label up to and including the passed label.
+
+        For sorted indexes, return the index label that is the latest among
+        the labels that are not later than the passed index label.
```

Comment:
I find this whole docstring a bit confusing to be honest. I think it's a complex method to document, but I think we can do better.

For the examples, I find this much clearer:
```
import pandas

idx = pandas.Index(['a', 'b', 'd', 'e'])
idx.asof('c')
```

For the description, it's difficult, but something like "Return the label from the index, or the previous if not found" would be something easier to understand for me.

For the rest looks quite good.

Reply:
It's hard to be completely understandable within a single line. But I'll try again using your suggestions. Thanks!